ChopChop/Storage/StorageManager.swift

```diff
@@ -17,13 +17,32 @@ struct StorageManager {
         var ingredientRecords = recipe.ingredients.map { ingredient in
             RecipeIngredientRecord(recipeId: recipe.id, name: ingredient.name, quantity: ingredient.quantity.record)
         }
-        var stepRecords = recipe.steps.enumerated().map { index, step in
-            RecipeStepRecord(recipeId: recipe.id, index: index + 1, content: step.content)
+        var graphRecord = RecipeStepGraphRecord(recipeId: recipe.id)
+        var stepRecords = recipe.stepGraph.nodes.map { node in
+            RecipeStepRecord(graphId: recipe.stepGraph.id, content: node.label.content)
         }
-
-        try appDatabase.saveRecipe(&recipeRecord, ingredients: &ingredientRecords, steps: &stepRecords)
+        var edgeRecords = recipe.stepGraph.edges.map { _ in
+            RecipeStepEdgeRecord(graphId: recipe.stepGraph.id)
+        }
```

Comment:
It looks like you're actually passing a non-empty array to `AppDatabase` here, but in the associated method, it looks like you're overriding this array entirely.